### PR TITLE
[r] Update documentation on g++ requirement for R package

### PR DESCRIPTION
**Pull request by CharlesBordet**
**Issue and/or context:**

Issue #2029 on updating documentation to mention the version of C++ compiler to install the R package

**Changes:**

Update `aps/r/README.md` 

**Notes for Reviewer:**

N/A

### CODE REVIEW

File: apis/r/README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ install.packages('tiledbsoma', repos = c('https://tiledb-inc.r-universe.dev',
                                          'https://cloud.r-project.org'))
 ```
 
-Installing from source on Unix-like platforms requires `cmake` and `git`.
+Installing from source on Unix-like platforms requires `cmake`, `git`, and a recent version of a C++ compiler (it was tested on `g++` version 10.5.0).
```
Comment (eddelbuettel) [inline apis/r/README.md:17]: Very nice.   Maybe reword to 

```suggestion
Installing from source on Unix-like platforms requires `cmake`, `git`, and a recent-enough C++ compiler (`g++` version 10.* or above).
```

We may want to add the `clang++` version too but I am unsure which it is. Maybe `clang++-16` ?

**Comment by CharlesBordet** [comment]
Thanks for the suggestion, I have committed your rewriting.

About clang++: I don't know what version is required, I didn't have to install it specifically, nor can I actually find it (`clang` or `clang++` is not found).

**Review by eddelbuettel — APPROVED** [review]
That looks good to me!  

Thank you for helping to make our documentation better!
